here-naksha-app-service/src/main/java/com/here/naksha/app/service/http/ops/SpatialUtil.java

```diff
@@ -32,13 +32,15 @@ public class SpatialUtil {
 
   private SpatialUtil() {}
 
-  public static @NotNull SOp buildOperationForTile(final @NotNull String tileType, final @NotNull String tileId) {
+  public static @NotNull SOp buildOperationForTile(
+      final @NotNull String tileType, final @NotNull String tileId, final int margin) {
     try {
       if (!TILE_TYPE_QUADKEY.equals(tileType)) {
         throw new XyzErrorException(XyzError.ILLEGAL_ARGUMENT, "Tile type " + tileType + " not supported");
       }
-      final Geometry geo =
-          WebMercatorTile.forQuadkey(tileId).getAsPolygon().getGeometry();
+      final Geometry geo = WebMercatorTile.forQuadkey(tileId)
+          .getExtendedBBoxAsPolygon(margin)
+          .getGeometry();
```

Comment:
Is this chain of calls NPE-proof? Do `forQuadkey(tileId)` and `getExtendedBBoxAsPolygon(margin)` guarantee to return non-null values?

Reply:
as per logic, no, but have adjusted code to handle other exceptions as well to be safe.